src/test/java/sysc4806/project24/mini_survey_monkey/integration/ControllerIntegrationTest.java

```diff
@@ -199,5 +200,25 @@ public void testClosingSurvey() throws Exception {
                     });
         });
     }
+
+    @Test
+    public void testNavigatingToSharingLink() throws Exception {
```

Comment:
This can be added in a different PR, but we should ensure that only "Open" surveys can be accessed via the "/r/id" link

Reply:
I second this. 

This should be an easy thing to implement in the interceptors in a future PR.